ui/tests/integration/components/edit-form-kmip-role-test.js

```diff
@@ -0,0 +1,182 @@
+import { later, run } from '@ember/runloop';
+import { resolve } from 'rsvp';
+import EmberObject, { computed } from '@ember/object';
+import Service from '@ember/service';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, settled, click } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+import engineResolverFor from 'ember-engines/test-support/engine-resolver-for';
+const resolver = engineResolverFor('kmip');
+
+const flash = Service.extend({
+  success: sinon.stub(),
+});
+const namespace = Service.extend({});
+
+const createModel = options => {
+  let model = EmberObject.extend({
+    fields: computed('newFields', function() {
+      return this.newFields.map(field => ({ name: field, type: 'boolean' }));
+    }),
+    /* eslint-disable ember/avoid-leaking-state-in-ember-objects */
+    newFields: ['operationAll', 'operationNone', 'operationGet', 'operationCreate', 'operationDestroy'],
+    destroyRecord() {
+      return resolve();
+    },
+    save() {
+      return resolve();
+    },
+    rollbackAttributes() {},
+  });
+  return model.create({
+    ...options,
+  });
+};
+
+module('Integration | Component | edit form kmip role', function(hooks) {
+  setupRenderingTest(hooks, { resolver });
+
+  hooks.beforeEach(function() {
+    run(() => {
+      this.owner.unregister('service:flash-messages');
+      this.owner.register('service:flash-messages', flash);
+      this.owner.register('service:namespace', namespace);
+    });
+  });
+
+  test('it renders: new model', async function(assert) {
+    let model = createModel({ isNew: true });
+    this.set('model', model);
+    await render(hbs`<EditFormKmipRole @model={{model}} />`);
+
+    assert.dom('[name=role-display]:checked').hasValue('operationAll', 'defaults to all on new models');
+  });
+
+  test('it renders: operationAll', async function(assert) {
+    let model = createModel({ operationAll: true });
+    this.set('model', model);
+    await render(hbs`<EditFormKmipRole @model={{model}} />`);
+
+    assert.dom('[name=role-display]:checked').hasValue('operationAll', 'sets operationAll');
+  });
+
+  test('it renders: operationNone', async function(assert) {
+    let model = createModel({ operationNone: true, operationGet: true });
+    this.set('model', model);
+    await render(hbs`<EditFormKmipRole @model={{model}} />`);
+
+    assert.dom('[name=role-display]:checked').hasValue('operationNone', 'sets operationNone');
```

Comment:
if `operationNone` is true, will that override `operationGet`? if that's the case, should we also test that `operationGet` is not checked?

Reply:
I should probably remove operationGet here because the data should never be returned like that from the server, but the init fn does prioritize `operationNone` and `operationAll` over the others in the case where data did come in that way. Within a group of radio buttons that share the same name, there will only be one item that's checked, so I don't think we need to test that "choose" isn't checked.